### PR TITLE
Abstract and refactor error handling

### PR DESCRIPTION
**Pull request by simao**

### CODE REVIEW

File: common/src/main/scala/org/genivi/sota/http/ErrorHandler.scala [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  private val onMissingEntity: PF = {
+    case MissingEntity(name) =>
+      complete(StatusCodes.NotFound ->
+        ErrorRepresentation(ErrorCodes.MissingEntity, s"${name.getSimpleName} not found"))
+  }
+
+  private val onConflictingEntity: PF = {
+    case EntityAlreadyExists(name) =>
+      complete(StatusCodes.NotFound ->
```
Comment (Danten) [inline common/src/main/scala/org/genivi/sota/http/ErrorHandler.scala]: Should this be StatusCodes.Conflict ?

Comment (simao) [inline common/src/main/scala/org/genivi/sota/http/ErrorHandler.scala]: yes :)

File: external-resolver/src/main/scala/org/genivi/sota/resolver/packages/PackageDirectives.scala [hunk 2]
```diff
@@ -50,27 +49,13 @@ class PackageDirectives(namespaceExtractor: Directive1[Namespace])
...
+    complete(db.run(PackageFilterRepository.addPackageFilter(PackageFilter(ns, id.name, id.version, fname))))
 
   def deletePackageFilter(ns: Namespace, id: PackageId, fname: String Refined Filter.ValidName): Route =
-    completeOrRecoverWith(db.run(
-      PackageFilterRepository.deletePackageFilter(PackageFilter(ns, id.name, id.version, fname)))) {
-        case PackageFilterRepository.MissingPackageFilterException =>
-          complete(StatusCodes.NotFound ->
-            ErrorRepresentation( ErrorCode("filter_not_found"),
-              s"No filter for namespace '$ns' with the name '${fname.get}'" +
-              s"defined for package ${id.name}-${id.version}"))
```
Comment (Danten) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/packages/PackageDirectives.scala]: It seems that we will not only print "No filter with defined for package" which seems less helpful. Am I right that we will have a less informative error message? Does that matter?

Comment (simao) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/packages/PackageDirectives.scala]: The route where this is used is sending this information already so I don't see a reason to keep it...

Comment (Danten) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/packages/PackageDirectives.scala]: Fair enough.

**Comment by Danten** [comment]
Review Score: 11/11

**Pull request merged**